### ISSUE TITLE
Doxygen fails to parse call references in C++ methods with throw specifications (Origin: bugzilla #642668)

### ISSUE THREAD

**Issue by doxygen** [doxygen/doxygen#4165]
status NEW severity *minor* in component *general* for *---*
Reported in version *1.8.2-SVN* on platform *Other*
Assigned to: Dimitri van Heesch

Original attachment names and IDs:
- _[throw.cpp](https://sourceforge.net/projects/doxygen/files/attachments/bug642668_181206_throw.cpp/download/download)_ (ID 181206)
- _[Doxyfile](https://sourceforge.net/projects/doxygen/files/attachments/bug642668_181207_Doxyfile/download/download)_ (ID 181207)
- _[xml.tgz](https://sourceforge.net/projects/doxygen/files/attachments/bug642668_181208_xml.tgz/download/download)_ (ID 181208)

On 2011-02-18 13:46:54 +0000, A. Saratow wrote:

    Created attachment 181206
    Test source file
    
    For methods with throw specifications doxygen seems sometimes to fail to
    parse the call references properly. For example some methods will contain no
    'calls' or 'called by' graphs, although this option is enabled in the
    Doxyfile for the complete solution. At some other places the calls are mis-
    linked to other classes in the case the methods have the same names.
    
    See attached xml.tgz archive, e.g. the classTable.xml file:
    Method Table::test3() is linked to Bable::callA() and Bable::callB() instead
    of Table::callA() and Table::callB (sorry for the stupid names).
    
    Steps to reproduce:
    Generate decumentation from the supplied source file and Doxyfile
    configuration.
    Doxygen was built on a x86 Ubuntu 10.10 system (default settings) from
    revision 755.

On 2011-02-18 13:47:28 +0000, A. Saratow wrote:

    Created attachment 181207
    Doxygen configuration file

On 2011-02-18 13:48:09 +0000, A. Saratow wrote:

    Created attachment 181208
    Doxygen XML output

On 2011-02-24 11:57:15 +0000, A. Saratow wrote:

    who the heck needs throw specifications :)

On 2012-11-18 11:07:39 +0000, Dimitri van Heesch wrote:

    Changed version 'latest' to '1.8.2-SVN' so I can remove 'latest' as an
    option as it is a moving target.

**Comment by albert-github** [comment]
Running the given example with the doxygen version 1.9.4 we see (in the HTML version):

![image](https://user-images.githubusercontent.com/5223533/183083824-08da269b-abd9-4dac-8737-28a0ce367263.png)

with the current master (1.9.5 (5763255ba4fb02b7fa11300515f46fbfda38180e)) we see:

![image](https://user-images.githubusercontent.com/5223533/183084025-d0322346-0f77-4a58-812e-f82ed633d3b8.png)


Analog in the XML versions we see;
**1.9.4** version:
```
        <references refid="classphat_1_1_bable_1a7386b50fdef8f61a534de43bff86290f" compoundref="throw_8cpp" startline="108" endline="108">phat::Bable::callA</references>
        <references refid="classphat_1_1_bable_1a90e25ea583af907362a22e574a637942" compoundref="throw_8cpp" startline="109" endline="109">phat::Bable::callB</references>
```
and in the master version:
```
        <references refid="classphat_1_1_table_1a43e1aa987854f60c390dbbfa166f364d" compoundref="throw_8cpp" startline="238" endline="238">callA</references>
        <references refid="classphat_1_1_table_1a299ca3d1edd4708838fe85fdf84dfb55" compoundref="throw_8cpp" startline="239" endline="239">callB</references>
```

In the 1.9.4 version the reference is still incorrect, but in the master version the right references are given.

I think the problem is fixed but not released.

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.5.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
